File: 0101-symmetric-tree/0101-symmetric-tree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    bool helper(TreeNode* A, TreeNode* B)
    {
        if(A==NULL && B==NULL)
            return true;
         if(A==NULL || B==NULL || A->val != B->val)
            return false;  
         return helper(A->left,B->right) && helper(A->right,B->left);     
    }
    bool isSymmetric(TreeNode* root) {
        if(root ==NULL)
            return true;
        return helper(root->left,root->right);
             
    }
};

File: 0101-symmetric-tree/0101-symmetric-tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def isSymmetric(self, root: Optional[TreeNode]) -> bool:
        def sym(p,q):
            if not p and not q:
                return True
            if (p and not q) or (q and not p):
                return False
            if  p.val == q.val:
                return sym(p.left,q.right) and sym(p.right , q.left)
            return False    
        if not root:
            return True
        return sym(root.left , root.right)


        